File: src/main/java/ar/com/flow/bankaccount/transfer/DifferentAccounts.java
package ar.com.flow.bankaccount.transfer;

import ar.com.flow.bankaccount.BankAccount;
import ar.com.flow.bankaccount.transaction.Precondition;

public class DifferentAccounts implements Precondition {
    private final BankAccount debitAccount;
    private final BankAccount creditAccount;

    public DifferentAccounts(BankAccount debitAccount, BankAccount creditAccount) {
        this.debitAccount = debitAccount;
        this.creditAccount = creditAccount;
    }

    public void check() {
        if (debitAccount.equals(creditAccount)) {
            throw new SameAccountException();
        }
    }
}

File: src/main/kotlin/ar/com/flow/bankaccount/transfer/DifferentAccounts.kt
package ar.com.flow.bankaccount.transfer

import ar.com.flow.bankaccount.BankAccount
import ar.com.flow.bankaccount.transaction.Precondition

class DifferentAccounts(
    private val debitAccount: BankAccount,
    private val creditAccount: BankAccount
) : Precondition {

    override fun check() {
        if (debitAccount == creditAccount) {
            throw SameAccountException()
        }
    }
}